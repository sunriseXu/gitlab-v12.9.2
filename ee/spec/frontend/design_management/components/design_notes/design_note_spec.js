import { shallowMount } from '@vue/test-utils';
import DesignNote from 'ee/design_management/components/design_notes/design_note.vue';
import UserAvatarLink from '~/vue_shared/components/user_avatar/user_avatar_link.vue';
import TimeAgoTooltip from '~/vue_shared/components/time_ago_tooltip.vue';

const scrollIntoViewMock = jest.fn();
HTMLElement.prototype.scrollIntoView = scrollIntoViewMock;

const $route = {
  hash: '#note_123',
};

describe('Design note component', () => {
  let wrapper;

  const findUserAvatar = () => wrapper.find(UserAvatarLink);
  const findUserLink = () => wrapper.find('.js-user-link');

  function createComponent(props = {}) {
    wrapper = shallowMount(DesignNote, {
      propsData: {
        note: {},
        ...props,
      },
      mocks: {
        $route,
      },
    });
  }

  afterEach(() => {
    wrapper.destroy();
  });

  it('should match the snapshot', () => {
    createComponent({
      note: {
        id: '1',
        createdAt: '2019-07-26T15:02:20Z',
        author: {
          id: 'author-id',
        },
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render an author', () => {
    createComponent({
      note: {
        id: '1',
        author: {
          id: 'author-id',
        },
      },
    });

    expect(findUserAvatar().exists()).toBe(true);
    expect(findUserLink().exists()).toBe(true);
  });

  it('should render a time ago tooltip if note has createdAt property', () => {
    createComponent({
      note: {
        id: '1',
        createdAt: '2019-07-26T15:02:20Z',
        author: {
          id: 'author-id',
        },
      },
    });

    expect(wrapper.find(TimeAgoTooltip).exists()).toBe(true);
  });

  it('should trigger a scrollIntoView method', () => {
    createComponent({
      note: {
        id: 'gid://gitlab/DiffNote/123',
        author: {
          id: 'author-id',
        },
      },
    });

    expect(scrollIntoViewMock).toHaveBeenCalled();
  });
});
