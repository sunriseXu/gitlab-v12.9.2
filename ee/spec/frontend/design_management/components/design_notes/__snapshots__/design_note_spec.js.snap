// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Design note component should match the snapshot 1`] = `
<timeline-entry-item-stub
  class="design-note note-form"
  id="note_undefined"
>
  <user-avatar-link-stub
    imgalt=""
    imgcssclasses=""
    imgsize="40"
    imgsrc=""
    linkhref=""
    tooltipplacement="top"
    tooltiptext=""
    username=""
  />
   
  <a
    class="js-user-link"
    data-user-id="author-id"
  >
    <span
      class="note-header-author-name bold"
    >
      
    </span>
     
    <!---->
     
    <span
      class="note-headline-light"
    >
      @
    </span>
  </a>
   
  <span
    class="note-headline-light note-headline-meta"
  >
    <span
      class="system-note-message"
    />
     
    <span
      class="system-note-separator"
    />
     
    <a
      class="note-timestamp system-note-separator"
      href="#note_undefined"
    >
      <time-ago-tooltip-stub
        cssclass=""
        time="2019-07-26T15:02:20Z"
        tooltipplacement="bottom"
      />
    </a>
  </span>
   
  <div
    class="note-text md"
    data-qa-selector="note_content"
  />
</timeline-entry-item-stub>
`;
